v3-sdk/minting-position/src/libs/contracts.ts

```diff
@@ -0,0 +1,90 @@
+// This file contains code to easily connect to and get information from a wallet on chain
+
+import { Currency } from '@uniswap/sdk-core'
+import { ethers } from 'ethers'
+import { ERC20_ABI, NONFUNGIBLEPOSITIONMANAGER_ABI } from './constants'
+import { toReadableAmount } from './conversion'
+import {
+  NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS,
+  AMOUNT_TO_APPROVE,
+} from './constants'
+import { sendTransaction, TransactionState } from './providers'
+
+export async function getCurrencyBalance(
+  provider: ethers.providers.Provider,
+  address: string,
+  currency: Currency
+): Promise<string> {
+  // Handle ETH directly
+  if (currency.isNative) {
+    return ethers.utils.formatEther(await provider.getBalance(address))
+  }
+
+  // Get currency otherwise
+  const currencyContract = new ethers.Contract(
+    currency.address,
+    ERC20_ABI,
+    provider
+  )
+  const balance: number = await currencyContract.balanceOf(address)
+  const decimals: number = await currencyContract.decimals()
+
+  // Format with proper units (approximate)
+  return toReadableAmount(balance, decimals).toString()
+}
+
+export async function getPositionIds(
+  provider: ethers.providers.Provider,
+  address: string,
+  contractAddress: string
+): Promise<number[]> {
+  // Get currency otherwise
+  const positionContract = new ethers.Contract(
+    contractAddress,
+    NONFUNGIBLEPOSITIONMANAGER_ABI,
+    provider
+  )
+  // Get number of positions
+  const balance: number = await positionContract.balanceOf(address)
+
+  // Get all positions
+  const tokenIds = []
+  for (let i = 0; i < balance; i++) {
+    const tokenOfOwnerByIndex: number =
+      await positionContract.tokenOfOwnerByIndex(address, i)
+    tokenIds.push(tokenOfOwnerByIndex)
+  }
+
+  return tokenIds
+}
+
+export async function getTokenTransferApprovals(
+  provider: ethers.providers.Provider,
+  tokenAddress: string,
+  fromAddress: string,
+  toAddress: string
+): Promise<TransactionState> {
+  if (!provider) {
+    console.log('No Provider Found')
+    return TransactionState.Failed
+  }
+
+  try {
+    const tokenContract = new ethers.Contract(tokenAddress, ERC20_ABI, provider)
+
+    const transaction = await tokenContract.populateTransaction.approve(
+      toAddress,
+      AMOUNT_TO_APPROVE
+    )
+
+    await sendTransaction({
+      ...transaction,
+      from: fromAddress,
+    })
+  } catch (e) {
+    console.error(e)
+    return TransactionState.Failed
+  }
+
+  return TransactionState.Sent
+}
```

Comment:
These two feel very distinct from balance info - maybe a `positions.ts` library instead? `contracts`is a bit overgeneralized to me IMO

Reply:
I see! Technically, they are the same thing but use case wise, it does feel different